src/Database/LSMTree/Internal/BlobRef.hs

```diff
@@ -5,23 +5,35 @@
 module Database.LSMTree.Internal.BlobRef (
     BlobRef (..)
   , BlobSpan (..)
+  , blobRefSpanSize
+  , readBlob
+  , readBlobIOOp
   ) where
 
 import           Control.DeepSeq (NFData (..))
+import           Control.RefCount (RefCounter)
+import qualified Data.Primitive.ByteArray as P (MutableByteArray,
+                     newPinnedByteArray, unsafeFreezeByteArray)
 import           Data.Word (Word32, Word64)
+import qualified Database.LSMTree.Internal.RawBytes as RB
+import           Database.LSMTree.Internal.Serialise (SerialisedBlob (..))
+import qualified System.FS.API as FS
+import           System.FS.API (HasFS)
+import qualified System.FS.BlockIO.API as FS
 
 -- | A handle-like reference to an on-disk blob. The blob can be retrieved based
 -- on the reference.
 --
 -- See 'Database.LSMTree.Common.BlobRef' for more info.
-data BlobRef run = BlobRef {
-      blobRefRun  :: !run
-    , blobRefSpan :: {-# UNPACK #-} !BlobSpan
+data BlobRef m h = BlobRef {
+      blobRefFile  :: !h
```

Comment:
is there a reason why this is not `Handle h` directly?

Reply:
ok, it looks like other places don't have `Handle h`

Reply:
Yeah I wondered the same. I tried changing it but ran into inconsistencies with other places not using `Handle h`. I think one could consistently swap them all over.